### ISSUE TITLE
Full fine-tuning followed by the original paper

### ISSUE THREAD

**Issue by maxin-cn** [eisneim/ltx_lora_training_i2v_t2v#18]
Thanks for shareing the code of the LoRA fine-tuning.

After reviewing the code, I noticed that the released training code seems to only train the newly added LoRA parameters using flow matching. I’d like to inquire if you have any plans to reproduce the original training method from the LTX-Video paper, where the Transformer and VAE are trained jointly? Or do you think that LoRA training alone is sufficient for downstream tasks, thus the full training followed by the LTX-video paper is not necessary?

**Comment by eisneim** [comment]
I think the VAE in LTX is already pretty good considering the compression ratio 32x32x8 much much higher than Hunyuan or Wan, even better than CosmosTokenizer VAE, Microsoft's Reducio VAE has good compression ratio but has length limit;

if you want to do some other downstream task, modify cross attention add more control, you better choose Wan or Hunyuan, LTX is super compact and  2b size is quite limiting, and easy to collapse; 

unless you are aiming to deploy the model to production, speed and compute cost is your main concern

**Comment by maxin-cn** [comment]
Thank you for your reply. The inference speed is one of the things I consider. I think the reasoning speed Hunyuan or WAN2.1 is too slow.

**Comment by eisneim** [comment]
@maxin-cn i'm working on 0.9.5 official codebase now (not diffusers version), it's kinda working but need more test, if you are interested, add me on wechat: eisneim, i can share the code base

**Comment by Wyatt-t-t** [comment]
> [@maxin-cn](https://github.com/maxin-cn) i'm working on 0.9.5 official codebase now (not diffusers version), it's kinda working but need more test, if you are interested, add me on wechat: eisneim, i can share the code base

Hello, I saw that LTX has released version 0.95 for Diffusers last month. May I ask why it's not possible to directly use its weights for training based on the existing framework? Is it because the model architectures between the two versions are different, or is there some difference in the implementation?

**Comment by eisneim** [comment]
@Wyatt-t-t Diffusers version is slower, and result is slightly different

**Comment by StarsTesla** [comment]
@Wyatt-t-t I inference some, got different result between diffuser and offical repo, the diffuser version is worst.

**Comment by erenovic** [comment]
Hello @eisneim, I am very much interested in 0.9.5 version. May I help out with testing while trying it out for a small project?

**Comment by eisneim** [comment]
@erenovic sorry i forgot to check github notifications, the code is here: [https://github.com/eisneim/ltx_video_train_none_diffusers](https://github.com/eisneim/ltx_video_train_none_diffusers)

**Comment by erenovic** [comment]
Hello @eisneim , thank you for sharing the repository, I didn't realize it was already there :) 

In the meanwhile, I have been trying to use the keyframing that they introduced in version 0.9.5 but simply mapping the code from inference pipeline to training did not work well. I am trying to condition on the last frame as well but the conditioning quality degrades throughout training. Do you have any idea how additional keyframe conditioning might be integrated to the training? They seem to just concatenate the conditioning latents but I don't see how this leads to improved keyframes?

**Comment by eisneim** [comment]
@erenovic first + last frame condition is just like first frame condition, my code  use different method: initialize noisy latent with clean first fame and also provide a condition mask to prevent denoising the first frame; you can modify the code to add last frame; 

to get good results, i think you need to train tons of resolutions and frame length, to make sure the 3d RoPE learns how to handle position correctly, my code pre-compute many resolutions into latent data, then train the model with mixed resolution and frame length; i have also done some experiments on video to video translation to generate video bokeh using ltx 0.9.6

original iphone video:

https://github.com/user-attachments/assets/0f76a98e-56bf-44f9-a2e7-3cd0fc6ed55a

5 step generation on RTX 4090 48GB

https://github.com/user-attachments/assets/7101522f-de82-4a9e-ae06-3b5b9bd0ebe8
